### ISSUE TITLE
Income Limits: result-errors e2e test disabled as flaky

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#16736]
## Description

- Test in question: `src/applications/income-limits/tests/cypress/results-errors.cypress.spec.js`
- e2e flaky disabled report: https://depo-platform-documentation.scrollhelp.site/developer-docs/currently-disallowed-cypress-test-specs-on-va-gov
- Workflow: https://github.com/department-of-veterans-affairs/vets-website/actions/runs/7381549524

## History
[Oct 2023 thread](https://dsva.slack.com/archives/CBU0KDSB1/p1696263748763819) from previous time this same test went flaky. Went flaky before, we were never able to reproduce or figure out why. Plat eventually removed from the flaky list.

## ACs
- [ ] Test passes consistently and/or thread with Platform support has been opened for Plat investigation into why we can't repro the errors and test is removed from flaky list

**Comment by randimays** [comment]
Error message & video from Curt on the aforementioned thread: https://dsva.slack.com/archives/CBU0KDSB1/p1704739132490429?thread_ts=1696263748.763819&cid=CBU0KDSB1

**Comment by randimays** [comment]
I found the root cause for this. The results errors spec is used to test the various issues that can crop up providing results on the final page. This includes invalid zip, dependents or year. The spec had a hardcoded mock endpoint to force API failure, but it was targeting 2023. The tests themselves were going through the "current" flow, which is now year 2024. I changed the API mock to dynamically target the year now, and we should not see this particular issue again.

PR here: https://github.com/department-of-veterans-affairs/vets-website/pull/27412
